pcsx2/Achievements.cpp

```diff
@@ -1772,9 +1773,10 @@ void Achievements::ClientLoginWithPasswordCallback(int result, const char* error
 
 	// Store configuration.
 	Host::SetBaseStringSettingValue("Achievements", "Username", params->username);
-	Host::SetBaseStringSettingValue("Achievements", "Token", user->token);
 	Host::SetBaseStringSettingValue("Achievements", "LoginTimestamp", fmt::format("{}", std::time(nullptr)).c_str());
+	Host::Internal::GetSecretsSettingsLayer()->SetStringValue("Achievements", "Token", user->token);
```

Comment:
As you said in the PR, it would be a good idea to remove the old token here. I suspect calling DeleteValue on the old one here would make sense.